base_layer/wallet_ffi/src/lib.rs

```diff
@@ -5447,6 +5463,69 @@ pub unsafe extern "C" fn wallet_create(
     }
 }
 
+/// Retrieves the network and version of an app that last accessed the wallet database
+///
+/// ## Arguments
+/// `config` - The TariCommsConfig pointer
+/// `error_out` - Pointer to an int which will be modified to an error code should one occur, may not be null. Functions
+/// as an out parameter.
+/// ## Returns
+/// `*mut TariNetworkAndVersion` - Returns the pointer to the TariNetworkAndVersion that contains the network and
+/// version
+///
+/// # Safety
+/// The ```network_and_version_destroy``` method must be called when finished with a TariNetworkAndVersion to prevent a
+/// memory leak
+#[no_mangle]
+pub unsafe extern "C" fn wallet_get_network_and_version(
```

Comment:
This is making this way more complicated, as above, if we make it 2 calls, we can only make this, get version. 
And then return a *char. We need need to create and destroy a custom mapped struct.